esp-hal/src/twai/mod.rs

```diff
@@ -791,6 +791,11 @@ where
             .err_warning_limit()
             .write(|w| unsafe { w.err_warning_limit().bits(96) });
 
+        // Disable CLKOUT
```

Comment:
Please explain why you added this piece of code, not (or not only, I guess...) what it does. It's completely unobvious to me (as an uninformed reader) why this has any effect.

Also, do we want this done for all MCUs, when the issue only seems to affect C3? Slightly related, do we want to have a test case in our test suite for this? (I think yes!)

Reply:
Currently there is no support for using the CLKOUT functionality of the TWAI peripheral in `esp-hal`, so I do not think there is a reason to leave the clock divider active. This also matches the behaviour from the ESP-IDF, which in the default configuration also does not use the CLKOUT functionality.

In the default configuration `clkout_divider` is zero, and `clkout_gpio` is `TWAI_IO_UNUSED`. The `clkout_divider` configuration is passed from `twai_driver_install_v2` to `twai_hal_configure`, which in turn calls `twai_ll_set_clkout` (https://github.com/espressif/esp-idf/blob/master/components/driver/twai/twai.c#L475 and https://github.com/espressif/esp-idf/blob/master/components/hal/twai_hal.c#L64).

The function `twai_ll_set_clkout` is chip specific, but for all chips with a TWAI peripheral does the exact same thing when passed zero. It sets the `TWAI_CLOCK_OFF` bit and sets `TWAI_CD` to zero. See [esp32](https://github.com/espressif/esp-idf/blob/master/components/hal/esp32/include/hal/twai_ll.h#L860-L873), [esp32c3](https://github.com/espressif/esp-idf/blob/master/components/hal/esp32c3/include/hal/twai_ll.h#L758-L771), [esp32c6](https://github.com/espressif/esp-idf/blob/master/components/hal/esp32c6/include/hal/twai_ll.h#L778-L791), [esp32p4](https://github.com/espressif/esp-idf/blob/master/components/hal/esp32p4/include/hal/twai_ll.h#L799-L812), [esp32s2](https://github.com/espressif/esp-idf/blob/master/components/hal/esp32s2/include/hal/twai_ll.h#L761-L774) and [esp32s3](https://github.com/espressif/esp-idf/blob/master/components/hal/esp32s3/include/hal/twai_ll.h#L758-L771). So I think it makes sense to match this configuration.

Why exactly this fixes the issue with receiving on GPIO1, I do not know. In theory leaving the clock divider active should not influence the receive operation, but I guess it does. If `esp-hal` is going to support the CLKOUT functionality, then we should probably add a note about receiving on GPIO1 when CLKOUT is enabled.

As for the test case, could you point me in the right direction where to add the test?